### ISSUE TITLE
Fix scheduler/simulation discrepancies about activity representation as intervals

### ISSUE THREAD

**Issue by adrienmaillard** [NASA-AMMOS/aerie#1087]
### Checked for duplicates

Yes - I've already checked

### Is this a regression?

No - This is a new bug

### Version

develop

### Describe the bug

In Aerie simulation, there is no order between activities being simulated at the same time except with anchoring. In simulation, an activity can act at both its time boundaries. They can all be potentially considered closed-closed intervals. Let's take A and B such that B starts at the exact same time as A is ending.

<img src="https://github.com/NASA-AMMOS/aerie/assets/94930916/002e4054-d281-4926-83fa-1c9e9efdcc99" width=40% height=40%>

There are two issues : 
- The scheduler assumes that activities can be represented by closed-open intervals. They should be represented with closed-closed intervals. 
- The second part has to do with `CoexistenceGoal`:
Currently, when we use `startsAt` or `endsAt` temporal constraints for placing coexisting activities, we make them start/end at one of the anchor time bounds which result in the above undefined behavior in simulation. In the above AB example the last instructions of A (after its last `delay()`) could execute after the first instructions of B (before its first `delay()`). Let's say that A depletes a battery and that B checks the level of the battery to act accordingly, this could lead to some unwanted/inaccurate behavior.

To fix that in `CoexistenceGoal` there are several possible paths that are currently being discussed:
- Either we add `startsJustAfter(TimeAnchor)` `endsJustBefore(TimeAnchor)` operators that makes start/end activities at bounds+1 microseconds, bounds-1 microseconds etc.
- We modify `startsAt`, `endsAt` behavior so that coexistent activities start/end at bounds+1 microseconds, bounds-1 microseconds etc. 
- We use anchors. If A is present and B has to be placed at the end of A, anchor B to the end of A. But if B is present and A has to finish at the start of A, the only way of using anchor would be to change the anchoring of B. The end of A cannot be anchored in the future. 
- Enforce order depending on which activity started first.
- Inject `delay(0)` at either start or end of activity effect model. Some discussion points below:
  - similar to what APGEN does
  - What would be the effects on decomposition ? Could the delay only apply to top-level activities ? 
  - That would change the semantics of spawning: the first child would see concurrent effect which was intended not to be the case 
  - Do we really need concurrency ? APGEN seems to work with total order 

### Reproduction

No reproduction.

### System Info

```shell
Not relevant
```

### Severity

Moderate

**Comment by jmdelfa** [comment]
I would discard the second option proposed, as it is highly counterintuitive. The startsAt operator should have the same meaning as it does in plain English. Therefore, when specifying startsAt(x), it should mean “starts at x” and not “starts at x+1”. 
In addition, there is a third option that is missing in the text that needs to be considered: to keep startsAt with the appropriate meaning and to make sure internally in the back end, that the appropriate intervals are passed to the simulator. That is, if we have an interval [x,y), the simulator should receive [x,y-1]  (or y - 0.00..1 for whatever precision we require in Aerie)

**Comment by adrienmaillard** [comment]
After several discussions, we have concluded that this issue will be resolved by:
1) enforcing that activities are represented by closed-closed intervals in scheduling 
2) providing an easy way of specifying very small delays between activities 
3) providing an anchoring mechanism in scheduling to the user 

Items 3) is the subject of #663. Items 1) and 2) are the subject of this ticket.

**Comment by adrienmaillard** [comment]
Anchoring has allowed to close this.
